### DISCUSSION TITLE
Semantic Climate Reports

### DISCUSSION THREAD

**Discussion by petermr** [petermr/petermr#31] [Announcements]
# Semantic Climate Reports

(Inspired by @mitko 2022-04 who suggested the idea of making IPCC reports semantic. Thanks)

# Background
The UN IPCC has published critically important major reports in climate/change, the most recent in 2021/2022:
* https://www.ipcc.ch/report/ar6/wg1/
* https://www.ipcc.ch/report/ar6/wg2/
* https://www.ipcc.ch/report/ar6/wg3/

The main reports in each of these are ca 3000 pages of PDF and written (in English) for specialists with many technical terms and abbreviations/acronyms. 

We believe that many citizens over the whole world will want to read these. We therefore are creating Open Source technology to index, search and annotate these. We section the paper into HTML (and hopefully JATS-XML) and use wikidata.org to interpret and elaborate on terms in the text. We also extract figures and their captions.

## Note on copyright

We are not posting the originals but creating derivative transformative works by Text and Data Mining (TDM). This has been specifically legal for non-commercial research in the UK since 2014. 
We would be delighted to work with the IPCC if the results of this work are useful to them.

**Comment by petermr** [comment]
# ContentMine strategy

(ContentMine.org no longer exists as a legal entity but its a useful label)

**The primary goal is to extract knowledge from open documents (or closed documents where allowed), systematize it and make it semantic and usable/useful.**

The idea of annotating climate documents arose from discussions with Simon Worthington and we set up "OpenClimate" e.g.
* https://genr.eu/wp/open-climate/
* https://genr.eu/wp/open-climate-knowledge-100-oa-for-climate-change/ (PMR sets out motivation and principles)
* 
and we ran sessions in our (private) Slack.

We also had contributions from several volunteers and pooled knowledge at:
* https://github.com/petermr/climate (Petermr)
* https://github.com/OCKProject/climate (Simon)

We did this for some months and **developed generic tools**. When the pandemic hit we used the name "openVirus" and used the tools to extract knowledge for viral epidemics (see https://github.com/petermr/openVirus/blob/master/README.md ) . The design of the tools is generic so any advance for viruses, plant_chemistry or climate benefits all disciplines. The methodology is excellently explained by interns from the Indian National Institute for Plant Genomics (NIPGR). Here's a good intro by our wonderful alumnus Shweata Hegde. (It's about plants but could be about any multidisciplinary science).
* https://github.com/petermr/CEVOpen/blob/master/outreach/INYAS_SNH_open_science_4.pdf (Social aspects)
* https://www.youtube.com/watch?v=9G8cfrDZgp4&ab_channel=ShweataN.Hegde (20 minute video includes tech) (start at 3.00)

In essence we:
* build a community with members who join for internships; many stay on
* develop novel tools for information extraction from public documents
* work very closely with anyone Open, especially Wikimedia/Wikidata
* build dictionaries based on terms , providing a complete semantic environment through Wikidata
* use dictionaries to:
- annotate the documents
- search the documents
- build indexes and other navigational tools

We believe these lead to semantic tools for reading the literature Our current targets include:
* open access scientific articles
* governmental and other public reports (mainly PDF)

More speculatively we are extracting data from diagrams is a semi-automatic fashion with words, line- and bar-plots as targets.

**Comment by petermr** [comment]
# Initial work 


## develop code to transform IPCC-PDF into HTML
Fairly well advanced. See  https://github.com/petermr/pyami/tree/main/temp/html/ (does not display in github, needs downloading)

## transform HTML into JATS-XML (standard for (bio)-science
Harder. We have to deduce the paragraphs , sections, chapters from format and numbering. But do-able.

## transform JATS-XML to browsable "book"
Many commodity tools for this (e.g. "e-Pub")

## index words
Many standard tools (e.g. `solr` ) will do this. We also use word frequency analysis in https:/github.com/petermr/docanalysis.

## create Wikidata-enabled dictionary
Tools for extracting words and lookup exist. Will also add Wikidata synonyms, and non-EN translations if they exist.

## search with dictionaries
Shweata Hegde's `docanalysis` uses SpaCy and other tools for search

**Reply by AmbrineH** [reply]
Kindly direct me to the code written for the conversion of  **IPCC-PDF into TEXT**. Has it been written? or should I proceed with writing it?

**Comment by petermr** [comment]
# project/directories

Initially

https://github.com/petermr/semanticClimate/tree/main/ipcc/ar6/wg3

**Comment by rgieseke** [comment]
As i have commented in the Twitter thread as well I'm not sure if it's worth spending much work on the PDF conversion as all texts should eventually come out as HTML as well. In any case, lots of interesting things to be done with machine readable reports to produce e.g. better search and reading experiences (e.g. ebook formats).

**Comment by ananyas168** [comment]
### Work Details - Ananya
Hi guys, I am working on data that can be found in this folder containing the starting 300 pages of the Climate Report. 
**Dataset Link**: https://github.com/petermr/pyami/tree/main/temp/html

The code which I have used to extract abbreviations from the HTML files can be found in the link below:
https://github.com/ananyas168/petermr/blob/main/climate_abbreviation_extraction.ipynb

 The idea behind the code is as follows:

- Convert an HTML to raw text using BeautifulSoup

- Extracting abbreviations using the NLP tool based on the [Schwartz-Hearst algorithm](https://psb.stanford.edu/psb-online/proceedings/psb03/schwartz.pdf). ( Here is the link for the tool: https://github.com/philgooch/abbreviation-extraction/blob/develop/README.md)

- Saving the extracted raw text and the abbreviations in txt format.

**Comment by petermr** [comment]
# Chapters - Owners and Conversion

The current chapters are at 
https://github.com/petermr/semanticClimate/tree/main/ipcc/ar6/wg3

## current owners
* Chapter01 https://github.com/petermr/semanticClimate/tree/main/ipcc/ar6/wg3/Chapter01
* Chapter04 https://github.com/petermr/semanticClimate/tree/main/ipcc/ar6/wg3/Chapter04 (Mitigation and development pathways in the near- to mid-term) (Enakshi)
* Chapter06 Samarth
* Chapter07 Ambreen
* Chapter 15 Kaartik

**Comment by petermr** [comment]
Robert ,
Thanks for your input!
The tools we are developing are very generic and will work with any PDF so
no work is wasted. Progress has been excellent and we have a semi-automated
toolchain:
* PDF->Text
* Text->HTML
* extraction of tables (incomplete)
* extraction of figures (just started)
These will work for any scientific or technical/government publication

One of the first things we are creating is a glossary iof terms linked to
Wikidata (Shweata) and abbreviarions+expansion (Ananya).


On Sat, Jun 4, 2022 at 1:46 PM Robert Gieseke ***@***.***>
wrote:

> As i have commented in the Twitter thread as well I'm not sure if it's
> worth spending much work on the PDF conversion as all texts should
> eventually come out as HTML as well. In any case, lots of interesting
> things to be done with machine readable reports to produce e.g. better
> search and reading experiences (e.g. ebook formats).
>

We hope to convert the HTML->JATS-XML when it should be easy to create
eBooks.

We are currently working on one chapter each so if someone really wants to
get involved we can allocate chapters. As we progress the software will
develop communally!

P.


—
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/31#discussioncomment-2882818>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS3AJYU4M6NGYYFGAW3VNNFY7ANCNFSM5WU4VSWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Reply by rgieseke** [reply]
That's great to hear and I didn't intend to belittle the great work being done in PDF extraction.

With regards to climate a great thing to parse would be the PDF based NDCs, the national climate plans.

I just updated my scraper of the list of NDCs at https://github.com/openclimatedata/ndcs

Being able to search these documents, have tables and graphs readable would be very valuable to research and NGOs assessing these.

There have been some efforts, e.g. at ClimateWatch but nothing systematic I think.

**Comment by petermr** [comment]
This is great!

On Wed, Jun 8, 2022 at 11:57 AM Robert Gieseke ***@***.***>
wrote:

> That's great to hear and I didn't intend to belittle the great work being
> done in PDF extraction.
>
Not at all. We always hope that others will create something better. But we
start out on the journey. This is a brilliant opportunity

> With regards to climate a great thing to parse would be the PDF based
> NDCs, the national climate plans.
>
> I just updated my scraper of the list of NDCs at
> https://github.com/openclimatedata/ndcs
>

WOW!
A) there are several languages. That's good. Is the list controlled? We can
use Wikidata to translate terms

> Being able to search these documents, have tables and graphs readable
> would be very valuable to research and NGOs assessing these.
>
I am optimistic of tables from the UN/IPCC itself as they all seem to use
the same tool/s. If the countries all use different tools there will be a
lot of variation. We'll see what happens.

> There have been some efforts, e.g. at ClimateWatch but nothing systematic
> I think.
>
This is a wonderful idea. We will try to work so that our tools are
reasonably distributable, so they can be offered to language-specific
volunteers. (We build the tools, but we don't run production).

P.

—
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/31#discussioncomment-2905080>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCSYW3AFK7DG5UD7M5H3VOB4A5ANCNFSM5WU4VSWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by rgieseke** [comment]
> Is the list controlled? 

I'm not sure I understand the question, NDCs are continuously updated and parties can submit in any of the official UN languages. Sometimes translations are provided, sometimes the 'official' primary document is provided in multiple languages.

I made the CSV version to see updates and be able to quickly download all the files.

> I am optimistic of tables from the UN/IPCC itself as they all seem to use
the same tool/s. If the countries all use different tools there will be a
lot of variation. We'll see what happens.

The NDCs vary greatly, every country uses their own custom formats.

WRI has text/HTML versions (also available on their website) of the NDC documents: https://github.com/WRI-ClimateWatch/ndc

https://www.climatewatchdata.org/ndcs-explore

For some assessments of the NDCs (which might point to relevant nomenclature and terms), see here: https://github.com/openclimatedata/ndc-assessments

The main question is to what level of emissions do these commitments add up to, but looking for topics like finance, equity, renewable energy plans is also quite relevant.

**Comment by petermr** [comment]
This has excited me - I will find time to download and analyze.

On Wed, Jun 8, 2022 at 8:01 PM Robert Gieseke ***@***.***>
wrote:

> Is the list controlled?
>
> I'm not sure I understand the question, NDCs are continuously updated and
> parties can submit in any of the official UN languages. Sometimes
> translations are provided, sometimes the 'official' primary document is
> provided in multiple languages.
>
> I made the CSV version to see updates and be able to quickly download all
> the files.
>
> I am optimistic of tables from the UN/IPCC itself as they all seem to use
> the same tool/s. If the countries all use different tools there will be a
> lot of variation. We'll see what happens.
>
> The NDCs vary greatly, every country uses their own custom formats.
>
> WRI has text/HTML versions (also available on their website) of the NDC
> documents: https://github.com/WRI-ClimateWatch/ndc
>
> https://www.climatewatchdata.org/ndcs-explore
>
> For some assessments of the NDCs (which might point to relevant
> nomenclature and terms), see here:
> https://github.com/openclimatedata/ndc-assessments
>
> The main question is to what level of emissions do these commitments add
> up to, but looking for topics like finance, equity, renewable energy plans
> is also quite relevant.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/31#discussioncomment-2908270>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS2FJLPPMBDISNJ3DR3VODUZJANCNFSM5WU4VSWA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by petermr** [comment]
# UN IPCC documents as learning objects.

We have been invited to pitch for a session at the Piaget Hackathon 16/18 June. https://hackathon-piaget.sparkboard.com/public . The overall theme is "Rethinking Education for the Twentyfirst Century". I believe that our work is fundamentally educational and can be expressed as the development of https://en.wikipedia.org/wiki/Learning_object s . The IPCC documents (AR6/WG1/2/3) are possibly the most important in the world and they should be used for the world to learn about climate change and contribute to the discussion.

We believe there are many citizens (at school and wider society) who want to know the messages in these documents (10000 pages PDF) and are fully capable of understanding them **if they are in more readable form** . They would then be usable in (real/virtual) classrooms, policy discussions and much more. Our project, driven by students (masters, undergraduate, school) has developed technology to bring the IPCC reports to life and make them reusable. A key strategy is to integrate them with Wikipedia and Wikidata, a collection of over 100 million metadata objects. 

Among the current barriers we hope to overcome are:
* PDF (inaccessible to unsighted people) - so convert to HTML
* Language. The primary language is English and there are few/no translations. Wikidata allows us to provide language equivalents for terms and links to non-English Wikipedia pages
* Jargon and Acronyms. Even for English speakers much of the language is based on technical phrases and acronyms. Wikidata can translate this into concepts citizens can more easily understand.
* Data in tables and images. This is not reusable (you can't put it in a spreadsheet or other viewer. We have tools to help convert these.
* Navigation. The flat PDF means it's impossible to navigate without painfully searching for every link. We intend to hyperlink the documents.

The resulting semantic objects can then be re-used with modern tools:
* summarisation
* hyperlinking
* aggregation 
* annotation
* searching and indexing

# Proposed hack
* we welcome everyone and look to integrate their contributions. Everything is fully Open
* Particular skills which will be valuable
  - climate science and politics (familiarity with IPCC!)
  - Wikimedia
  - learning objects and their use
  - Python , including in schools and offline
  - languages
  - story telling (text, graphics)
  - data and interactivity
* individuals/pairs choose a Chapter of IPCC/AR6/WG3 that interests them. Ca. 200 pp.
- run pyami and extract the text (or hopefully HTML)
- run docanalysis and find the commonest terms and phrases in the chapter.
- automatically look those up in Wikidata and create a dictionary
- use the dictionary to annotate the chapter text with hyperlinks
- (ambitious) extract/organize  tables from the chapter
- (ambitious) extract/organize figures from the chapter

I have run hacks before and can help to fit this into your timetable.

**Comment by ShweataNHegde** [comment]
### Details we need to make the pitch

1. Project Name
2. Project Summary
3. Image
4. Challenge Description
5. Team
6. Looking For

**Reply by ShweataNHegde** [reply]
Let's make pointers for all sub-headings? Is anyone here good at images and would want to take that up?

**Reply by ShweataNHegde** [reply]
I've started a Google Doc: https://docs.google.com/document/d/1J9XgnmLDyt5uZJQlsWhL6Hw9hMA7gNMrR48_aYlDtqc/edit?usp=sharing. It's more easier to collaborate there.

**Reply by ShweataNHegde** [reply]
Link to register yourselves: https://hackathon-piaget.sparkboard.com/projects

**Comment by ShweataNHegde** [comment]
We've submitted our pitch! 

You can find the link here: https://hackathon-piaget.sparkboard.com/project/62a327036c6ee3003da01352

**Comment by petermr** [comment]
# Discord We now have a slot in Discord (similar to Slack)
https://discord.com/channels/985212033169518622/985215792989753364
